RocketChat/Rocket.Chat
# [FIX] "Parent channel or group" search in discussions' creation throws "Unexpected end of JSON input" error

## Proposed changes (including videos or screenshots)


- Use `encodeURIComponent()` to encode values received by `_generateQueryFromParams()`.



## Issue(s)

[Task - ClickUp](https://app.clickup.com/t/5q8naj)

## Steps to test or reproduce

Open **Create New Discussion** and type "#" in the "**Parent channel or group**" field. **_Expected behavior:_** no error should be displayed (and the search's result should be empty).

## Further comments

The JSON input would end before expected when using the "#" char because this could be interpreted as the URL's **fragment** part.
